File: subset sums.java
//User function Template for Java//User function Template for Java
class Solution{
    ArrayList<Integer> subsetSums(ArrayList<Integer> arr, int N){
        // code here
        ArrayList <Integer> ans = new ArrayList <Integer>();
        int i=0;
        int s=0;
        fun(arr,i,ans,s);
        return ans;
    
    }
    void fun(ArrayList<Integer> arr , int i , ArrayList<Integer> ans , int s){
        if(i>=arr.size()){
            ans.add(s);
            return;
        }
        fun(arr , i+1 , ans , s+arr.get(i));
        fun(arr , i+1, ans, s);
    }
        
    
}

File: Subset Sums.py
class Solution:
    def subsetSums(self, arr, N):
        # code here
        ans=[]
        i=0
        s=0
        self.fun(arr,i,s,ans)
        return ans
    def fun(self,arr,i,s,ans):
        if (i>=len(arr)):
            ans.append(s)
            return
        self.fun(arr,i+1,arr[i]+s,ans)
        self.fun(arr,i+1,s,ans)
        